### PR TITLE
Loop over all classes of matrices

### PR DESCRIPTION
**Pull request by anastasia21112**
## Suite Sparse Linear Solve Defaults

- In MatrixDepot.jmd, added a list that should download all the matrices and loop over them

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Need to add Regex to the Project/Manifest.

File: benchmarks/LinearSolve/MatrixDepot.jmd [hunk 1]
```diff
@@ -26,20 +26,34 @@ algnames_transpose = reshape(algnames, 1, length(algnames))
 
 cols = [:red, :blue, :green, :magenta, :turqoise] # one color per alg
 
-matrices = ["HB/1138_bus", "HB/494_bus", "HB/662_bus", "HB/685_bus", "HB/bcsstk01", "HB/bcsstk02", "HB/bcsstk03", "HB/bcsstk04", 
-            "HB/bcsstk05", "HB/bcsstk06", "HB/bcsstk07", "HB/bcsstk08", "HB/bcsstk09", "HB/bcsstk10", "HB/bcsstk11", "HB/bcsstk12",
-            "HB/bcsstk13", "HB/bcsstk14", "HB/bcsstk15", "HB/bcsstk16"]
+# matrices = ["HB/1138_bus", "HB/494_bus", "HB/662_bus", "HB/685_bus", "HB/bcsstk01", "HB/bcsstk02", "HB/bcsstk03", "HB/bcsstk04",  "HB/bcsstk05", "HB/bcsstk06", "HB/bcsstk07", "HB/bcsstk08", "HB/bcsstk09", "HB/bcsstk10", "HB/bcsstk11", "HB/bcsstk12", "HB/bcsstk13", "HB/bcsstk14", "HB/bcsstk15", "HB/bcsstk16"]
```
Comment (ChrisRackauckas) [inline benchmarks/LinearSolve/MatrixDepot.jmd:29]: keep this around commented for testing, it's still useful but not for the big runs

File: benchmarks/LinearSolve/MatrixDepot.jmd [hunk 2]
```diff
@@ -26,20 +26,34 @@ algnames_transpose = reshape(algnames, 1, length(algnames))
...
+allmatrices_md = listnames("*/*")
+using Regex
```
Comment (ChrisRackauckas) [inline benchmarks/LinearSolve/MatrixDepot.jmd:31]:
```suggestion
```

**Comment by ChrisRackauckas** [comment]
Looks like it has an error in the file reading setup?

**Comment by ChrisRackauckas** [comment]
`listnames` doesn't give a string it gives a Markdown.MD. You'll want to convert those to a string.

**Comment by ChrisRackauckas** [comment]
I put a size limit in there to see if we can get a version to finish.

File: benchmarks/LinearSolve/MatrixDepot.jmd [hunk 3]
```diff
@@ -26,25 +26,32 @@ algnames_transpose = reshape(algnames, 1, length(algnames))
...
         rng = MersenneTwister(123)
-        A = mdopen(matrices[z]).A
+        A = mdopen(currMTX).A
         A = convert(SparseMatrixCSC, A)
         n = size(A, 1)
         matrix_size[z] = n
         percentage_sparsity[z] = length(nonzeros(A)) / n^2
         @info "$n × $n"
+
+        n > 100_000 && error("Skipping too large matrices")
```
Comment (ChrisRackauckas) [inline benchmarks/LinearSolve/MatrixDepot.jmd:53]:
```suggestion
        n > 10_000 && error("Skipping too large matrices")
```

File: benchmarks/LinearSolve/MatrixDepot.jmd [hunk 4]
```diff
@@ -26,25 +26,32 @@ algnames_transpose = reshape(algnames, 1, length(algnames))
...
         rng = MersenneTwister(123)
-        A = mdopen(matrices[z]).A
+        A = mdopen(currMTX).A
         A = convert(SparseMatrixCSC, A)
         n = size(A, 1)
         matrix_size[z] = n
         percentage_sparsity[z] = length(nonzeros(A)) / n^2
         @info "$n × $n"
+
+        n > 10_000 && error("Skipping too large matrices")
```
Comment (ChrisRackauckas) [inline benchmarks/LinearSolve/MatrixDepot.jmd:53]:
```suggestion
        n > 5_000 && error("Skipping too large matrices")
```

File: benchmarks/LinearSolve/MatrixDepot.jmd [hunk 5]
```diff
@@ -26,25 +26,32 @@ algnames_transpose = reshape(algnames, 1, length(algnames))
...
-            "HB/bcsstk13", "HB/bcsstk14", "HB/bcsstk15", "HB/bcsstk16"]
+# matrices = ["HB/1138_bus", "HB/494_bus", "HB/662_bus", "HB/685_bus", "HB/bcsstk01", "HB/bcsstk02", "HB/bcsstk03", "HB/bcsstk04",  "HB/bcsstk05", "HB/bcsstk06", "HB/bcsstk07", "HB/bcsstk08", "HB/bcsstk09", "HB/bcsstk10", "HB/bcsstk11", "HB/bcsstk12", "HB/bcsstk13", "HB/bcsstk14", "HB/bcsstk15", "HB/bcsstk16"]
+allmatrices_md = listnames("*/*")
 
-times = fill(NaN, length(matrices), length(algs))
-percentage_sparsity = fill(NaN, length(matrices))
-matrix_size = fill(NaN, length(matrices))
+times = fill(NaN, length(allmatrices_md.content[1].rows), length(algs))
+percentage_sparsity = fill(NaN, length(allmatrices_md.content[1].rows))
+matrix_size = fill(NaN, length(allmatrices_md.content[1].rows))
```
Comment (ChrisRackauckas) [inline benchmarks/LinearSolve/MatrixDepot.jmd:34]:
```suggestion
@info "Total number of matrices: $(allmatrices_md.content[1].rows)"
times = fill(NaN, length(allmatrices_md.content[1].rows), length(algs))
percentage_sparsity = fill(NaN, length(allmatrices_md.content[1].rows))
matrix_size = fill(NaN, length(allmatrices_md.content[1].rows))
```

File: benchmarks/LinearSolve/MatrixDepot.jmd [hunk 6]
```diff
@@ -100,6 +118,14 @@ p = scatter(matrix_size, times;
     legend = :outertopright)
 ```
 
+```julia
+algnames
+```
+
+```julia
+times
+```
```
Comment (ChrisRackauckas) [inline benchmarks/LinearSolve/MatrixDepot.jmd:127]:
```suggestion
```

**Comment by ChrisRackauckas** [comment]
Changing the plots to PNGs instead of vector graphics because they might just be too big to upload 😅

**Comment by ChrisRackauckas** [comment]
@staticfloat @thazhemadam can I get some help figuring out why after it generates the doc stuff it just stalls? I thought it would be because the plots were huge or something, but making them PNGs didn't do anything about it.

File: benchmarks/LinearSolve/MatrixDepot.jmd [hunk 7]
```diff
@@ -26,25 +26,33 @@ algnames_transpose = reshape(algnames, 1, length(algnames))
...
         rng = MersenneTwister(123)
-        A = mdopen(matrices[z]).A
+        A = mdopen(currMTX).A
         A = convert(SparseMatrixCSC, A)
         n = size(A, 1)
         matrix_size[z] = n
         percentage_sparsity[z] = length(nonzeros(A)) / n^2
         @info "$n × $n"
+
+        n > 5_000 && error("Skipping too large matrices")
```
Comment (ChrisRackauckas) [inline benchmarks/LinearSolve/MatrixDepot.jmd:54]:
```suggestion
        n > 1_000 && error("Skipping too large matrices")
```

**Comment by ChrisRackauckas** [comment]
I'm lowering the max matrix size to 1000x1000 to see if that makes a dent in that "upload time" that is unknown.

**Comment by staticfloat** [comment]
It's hard to tell; it looks to me like Weave just freezes.

File: benchmarks/LinearSolve/MatrixDepot.jmd [hunk 8]
```diff
@@ -26,25 +26,33 @@ algnames_transpose = reshape(algnames, 1, length(algnames))
...
         rng = MersenneTwister(123)
-        A = mdopen(matrices[z]).A
+        A = mdopen(currMTX).A
         A = convert(SparseMatrixCSC, A)
         n = size(A, 1)
         matrix_size[z] = n
         percentage_sparsity[z] = length(nonzeros(A)) / n^2
         @info "$n × $n"
+
+        n > 1_000 && error("Skipping too large matrices")
```
Comment (ChrisRackauckas) [inline benchmarks/LinearSolve/MatrixDepot.jmd:54]:
```suggestion
        n > 100 && error("Skipping too large matrices")
```

**Pull request merged**